### ISSUE TITLE
item_stock & item_type not load and save

### ISSUE THREAD

**Issue by rminformaticos** [jekkos/opensourcepos#1159]
on items form, those inputs doesnt save on DB if i select non-stock "checked" value.

and on edit item form, value doesnt load from DB.

(sorry for my bad english)

**Comment by SteveIreland** [comment]
Did you run the database scripts to add the `stock_type` and the `item_type` to the `ospos_items` table?  If so then something else might be going on .  I'm not able to replicate this issue.

Can you change the public/index.php file statement 56 from 

	define('ENVIRONMENT', isset($_SERVER['CI_ENV']) ? $_SERVER['CI_ENV'] : 'production');

to

	define('ENVIRONMENT', isset($_SERVER['CI_ENV']) ? $_SERVER['CI_ENV'] : 'testing');

and try it again?  Maybe there is a PHP error that's  being swallowed.

**Comment by rminformaticos** [comment]
hi, yes i have stock_type and item_type on ospos_items table.

change to development and same problem.

i test, change one item's stock_type value to 1 on DB, and it loads ok on edit item form, but when i save it, the data always save it to 0.

**Comment by odiea** [comment]
I cannot reproduce this issue.

**Comment by SteveIreland** [comment]
I just now downloaded a fresh master from jekkos and configured a clean database with a set of test items and all of the updates are working as expected.

Something else is going on that's preventing the changes from being saved.

Are ANY changes being saved to your `item` table?  What happens when you try to change the item description?

Also (not relevant to this issue but I noticed it and thought it should be highlighted), as @odiea shows in his test example, if you change an item that has inventory to non-stock the inventory will remain unchanged.  However sales will not automatically adjust inventory.
